trustfall_core/src/interpreter/hints/vertex_info.rs

```diff
@@ -194,7 +169,24 @@ impl<T: InternalVertexInfo + super::sealed::__Sealed> VertexInfo for T {
                 .collect::<Vec<RequiredProperty>>(),
         );
 
-        properties
+        properties.extend(current_component.vertices.values().flat_map(|v| {
+            v.filters
+                .iter()
+                .filter(|f| match f.right() {
+                    Some(Argument::Tag(FieldRef::ContextField(ctx))) => {
+                        current_vertex.vid == ctx.vertex_id
+                    }
+                    Some(Argument::Tag(FieldRef::FoldSpecificField(fsf))) => {
+                        current_vertex.vid == fsf.fold_root_vid
+                    }
+                    _ => false,
+                })
+                .map(|f| f.right().unwrap().as_tag().unwrap().field_name().into())
+                .map(RequiredProperty::new)
+                .collect::<Vec<RequiredProperty>>()
+        }));
+
+        Box::new(properties.into_iter())
```

Comment:
For now, this is totally fine. But it might be nice before merging to make this more lazy: we can avoid materializing everything into a `Vec` and instead chain the iterators (ones before `.collect()`) together and return that chained iterator.

Reply:
Ah, that is super cool! Thanks for the tip, changing it to use chain